docker-app/qfieldcloud/core/cron.py

```diff
@@ -35,3 +44,38 @@ def do(self):
         logger.info(
             f'Resend {len(invitation_emails)} previously failed invitation(s) to: {", ".join(invitation_emails)}'
         )
+
+
+class SetTerminatedWorkersToFinalStatusJob(CronJobBase):
+    # arbitrary number 3 here, it just feel a good number since the configuration is 10 mins
+    schedule = Schedule(run_every_mins=3)
+    code = "qfieldcloud.set_terminated_workers_to_final_status"
+
+    def do(self):
+
+        client = docker.from_env()
+        qgis_containers = client.containers.list(
+            sparse=True, filters={"ancestor": QGIS_CONTAINER_NAME}
+        )
+        qgis_container_ids = [c.id for c in qgis_containers]
```

Comment:
Btw didn't think of it before, but is this guaranteed to return all workers in the stack ? and not only on the current host ? Otherwise it will terminate perfectly fine containers !

I'm wondering if it wouldn't be safer to also check the `job.started_at` timestamp and only keep those that outlasted the timeout ?

Reply:
While this is pefrectly fine on composer setup, you are very right this could be dangerous on k8s setup where the containers are isolated from one another and this will return false negatives. Good catch #2.

Reply:
Should be good now.